packages/commonwealth/server/controllers/server_admin_methods/get_stats.ts

```diff
@@ -0,0 +1,119 @@
+import { Op, QueryTypes, WhereOptions } from 'sequelize';
+import { AppError } from '../../../../common-common/src/errors';
+import { CommunityAttributes } from '../../models/community';
+import { UserInstance } from '../../models/user';
+import { ServerAdminController } from '../server_admin_controller';
+
+export const Errors = {
+  NotAdmin: 'Must be a site admin',
+  CommunityNotFound: 'Community not found',
+};
+
+export type GetStatsOptions = {
+  user: UserInstance;
+  communityId?: string;
+};
+
+export type GetStatsResult = {
+  lastMonthNewCommunities: Array<{ id: string }>;
+  totalStats: {
+    numCommentsLastMonth: number;
+    numThreadsLastMonth: number;
+    numPollsLastMonth: number;
+    numReactionsLastMonth: number;
+    numProposalVotesLastMonth: number;
+    numMembersLastMonth: number;
+    numGroupsLastMonth: number;
+  };
+};
+
+export async function __getStats(
+  this: ServerAdminController,
+  { user, communityId }: GetStatsOptions,
+): Promise<GetStatsResult> {
+  if (!user.isAdmin) {
+    throw new AppError(Errors.NotAdmin);
+  }
+
+  // community is optional
+  let community: CommunityAttributes | undefined = undefined;
+  if (communityId) {
+    community = await this.models.Community.findByPk(communityId);
+    if (!community) {
+      throw new AppError(Errors.CommunityNotFound);
+    }
+  }
+
+  let lastMonthNewCommunities: Array<{ id: string }> = [];
+  if (!community) {
+    lastMonthNewCommunities = await this.models.sequelize.query(
+      `SELECT id FROM "Communities" WHERE created_at >= NOW() - INTERVAL '30 days'`,
+      { type: QueryTypes.SELECT },
+    );
+  }
+
+  const oneMonthAgo = new Date();
+  oneMonthAgo.setDate(oneMonthAgo.getDate() - 30);
+
+  const where = {
+    created_at: {
+      [Op.gte]: oneMonthAgo,
+    },
+  };
+  const whereChain: WhereOptions<{
+    created_at: any;
+    chain: string;
+  }> = {
+    ...where,
+  };
+  const whereCommunityId: WhereOptions<{
+    created_at: any;
+    community_id: string;
+  }> = { ...where };
+
+  if (community) {
+    whereChain.chain = communityId;
+    whereCommunityId.community_id = communityId;
+  }
+
+  const numCommentsLastMonth = await this.models.Comment.count({
```

Comment:
We can combine all of these queries into a single query if we use `UNION`. This will reduce network latency. If we don't want to do that we should parallelize all the queries with `Promise.all`

Reply:
The queries have been parallelized.

Regarding the error– that's odd. It shows fine for me locally. The UI has been updated so I'm not sure why that error would show. Maybe pull the latest, clear cache and restart server?

Otherwise, let me know what LOC the error occurs at.

<img width="725" alt="CleanShot 2024-01-02 at 18 51 48@2x" src="https://github.com/hicommonwealth/commonwealth/assets/126922418/cff0e7c3-dbac-4ddc-bd02-32ed495659fa">

Reply:
I reset DB, reset cache, and used a fresh server and still can't load the admin-panel. Not getting anything helpful in the console:
![Screenshot from 2024-01-03 20-44-57](https://github.com/hicommonwealth/commonwealth/assets/62490329/d4fc2d4c-9384-41de-a154-207701b9e5f1)

I'm deploying to Frack now to see if it happens there too.

Reply:
Happening on Frack too:
![Screenshot from 2024-01-03 20-58-42](https://github.com/hicommonwealth/commonwealth/assets/62490329/34763dc7-30d2-4331-83e1-653fc6839bde)
![Screenshot from 2024-01-03 20-58-33](https://github.com/hicommonwealth/commonwealth/assets/62490329/53da1ffa-2140-46e6-a3e8-b9f3e0cabbbc)

I reset the DB there as well and did a hard refresh.

Reply:
Sanity check: is your user a super user?

Reply:
Yea

Reply:
Fixed– thanks for testing!